p2panda-net/src/network.rs

```diff
@@ -166,8 +175,8 @@ where
 
     /// Sets the sync protocol for this network.
     ///
-    /// If a sync protocol is provided, a sync session will be completed before entering gossip
-    /// mode with any known peers with whom we share topics of interest.
+    /// If a sync protocol is provided, a sync session will be completed with any known peers with
+    /// whom we share topics of interest.
     pub fn sync(mut self, protocol: impl for<'a> SyncProtocol<'a, T> + 'static) -> Self {
```

Comment:
I could imagine that we soon want to be able to set sync protocol for a certain set of topics, that would be cool for supporting multiple applications sharing the same network. Just a reflection now....

Reply:
I meant that the you could register a sync protocol per topic (or collection of topic). The method signature might look like this:

```rust
pub fn sync(mut self, topics: Vec<T>, protocol: impl for<'a> SyncProtocol<'a, T> + 'static) -> Self {...}
```

Reply:
Nothing for this PR though of course :pray: